internal/session/session.go

```diff
@@ -351,7 +352,8 @@ func contains(ss []string, t string) bool {
 	return false
 }
 
-func newCert(ctx context.Context, wrapper wrapping.Wrapper, userId, jobId string, exp time.Time) (ed25519.PrivateKey, []byte, error) {
+// newCert creates a new session certificate. If addresses are supplied, they will be parsed and added to IP or DNS SANs as appropriate.
```

Comment:
should we allow addresses to be empty?

Reply:
I don't think it hurts anything. It's not always necessary. In practice we'll probably always supply something except in test code.

If you prefer I could ensure it's not empty and update tests, but I don't have a preference.